src/Mod/PartDesign/App/FeatureAddSub.cpp

```diff
@@ -64,6 +64,29 @@ short FeatureAddSub::mustExecute() const
     return PartDesign::Feature::mustExecute();
 }
 
+
+bool FeatureAddSub::onlyHasToRefine() const
+{
+    if( ! Refine.isTouched()){
+        return false;
+    }
+    if (rawShape.isNull()){
+        return false;
+    }
+    std::vector<App::Property*> propList;
+    getPropertyList(propList);
+    for (auto prop : propList){
+        if (prop != &Refine
+            /*&& prop != &SuppressedShape*/
+            && prop->isTouched()){
+            return false;
+        }
+    }
+    return true;

```

Comment:
Is this just `std::any_of`?

Reply:
Is this really clearer?
```cpp
    return std::none_of(propList.begin(),
                       propList.end(),
                       [this](App::Property* prop){
                           return prop == &Refine || prop->isTouched();
                       });
```

Reply:
For clarification the lambda function used for `std::none_of` does the opposite as the check inside the for loop.
`std::none_of` is  basically implemented as (see https://cplusplus.com/reference/algorithm/none_of)
```
while (first!=last) {
    if (pred(*first)) {
        return false;
    }
    ++first;
}
return true;
```

`FeatureAddSub::onlyHasToRefine()` can be simplified. At the top of the function it is already checked that `Refine` is not touched. So, in the next step it can be checked how many properties are touched. If the number is 1 it's only the Refine property.

So, it can be written as:
```
...
getPropertyList(propList);
return std::count_if(propList.begin(), propList.end(), [](auto prop) {
    return prop->isTouched();
}) == 1;
```

Reply:
> Is this really clearer?

Instead of an unnamed lambda you can use a named lambda to make the intention clear(er):
```
auto isOtherTouched = [this](App::Property* prop) {
    return (prop != &Refine && prop->isTouched());
};

return std::none_of(propList.begin(), propList.end(), isOtherTouched);
```